biogtr/datasets/sleap_dataset.py

```diff
@@ -0,0 +1,200 @@
+"""Module containing logic for loading sleap datasets."""
+import torch
+import imageio
+import sleap_io as sio
+from biogtr.datasets import data_utils
+from torch.utils.data import Dataset
+from torchvision.transforms import functional as tvf
+
+
+class SleapDataset(Dataset):
+    """Dataset for loading animal behavior data from sleap."""
+
+    def __init__(
+        self,
+        slp_files: list[str],
+        video_files: list[str],
+        padding: int = 5,
+        crop_size: int = 128,
+        chunk: bool = True,
+        clip_length: int = 500,
+        crop_type: str = "centroid",
+        mode: str = "train",
+        tfm: callable = None,
+        tfm_cfg: dict = None,
+    ):
+        """Initialize SleapDataset.
+
+        Args:
+            slp_files: a list of .slp files storing tracking annotations
+            video_files: a list of paths to video files
+            padding: amount of padding around object crops
+            crop_size: the size of the object crops
+            chunk: whether or not to chunk the dataset into batches
+            clip_length: the number of frames in each chunk
+            crop_type: `centroid` or `pose` - determines whether to crop around a centroid or around a pose
+            mode: `train` or `val`. Determines whether this dataset is used for training or validation.
+            Currently doesn't affect dataset logic
+            tfm: The augmentation function from albumentations
+            tfm_cfg: The config for the augmentations
+        """
+        self.slp_files = slp_files
```

Comment:
Should we handle directories here or at high level?

Reply:
assume handling at high level in train.py/inference.py etc